invenio_communities/assets/semantic-ui/js/invenio_communities/community/communitiesItems/CommunityCompactItemMobile.js

```diff
@@ -36,7 +37,7 @@ export const CommunityCompactItemMobile = ({
 
           <div className="flex align-items-center rel-mb-1">
             <a
-              href={links.self_html}
+              href={detailUrl || links.self_html}
```

Comment:
Based on the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR) we should be safe. If `detailUrl` can be converted to true it will return that value, otherwise it will check for `links.self_html`